pkg/alertmanager/multitenant_test.go

```diff
@@ -194,6 +228,102 @@ func TestMultitenantAlertmanager_loadAndSyncConfigs(t *testing.T) {
 	`), "cortex_alertmanager_config_last_reload_successful"))
 }
 
+func TestMultitenantAlertmanager_migrateStateFilesToPerTenantDirectories(t *testing.T) {
+	ctx := context.Background()
+
+	const (
+		user1 = "user1"
+		user2 = "user2"
+	)
+
+	store := prepareInMemoryAlertStore()
+	require.NoError(t, store.SetAlertConfig(ctx, alertspb.AlertConfigDesc{
+		User:      user2,
+		RawConfig: simpleConfigOne,
+		Templates: []*alertspb.TemplateDesc{},
+	}))
+
+	reg := prometheus.NewPedanticRegistry()
+	cfg := mockAlertmanagerConfig(t)
+	am, err := createMultitenantAlertmanager(cfg, nil, nil, store, nil, log.NewNopLogger(), reg)
+	require.NoError(t, err)
+
+	createFile(t, filepath.Join(cfg.DataDir, "nflog:"+user1))
+	createFile(t, filepath.Join(cfg.DataDir, "silences:"+user1))
+	createFile(t, filepath.Join(cfg.DataDir, "nflog:"+user2))
+	createFile(t, filepath.Join(cfg.DataDir, "templates", user2, "template.tpl"))
+
+	require.NoError(t, am.migrateStateFilesToPerTenantDirectories())
+	require.True(t, exists(t, filepath.Join(cfg.DataDir, user1, notificationLogSnapshot), false))
```

Comment:
I found the `true`/`false` confusing to read here, without looking at the function definition for `exists`, my brain interprets `exists(..., false)` as checking the file doesn't exist. Suggestion:

```
 	require.True(t, directoryExists(t, filepath.Join(cfg.DataDir, user1, notificationLogSnapshot)))
 	...
 	require.True(t, fileExists(t, filepath.Join(cfg.DataDir, user2, templatesDir)))
 ```
 
 Then a couple of helpers functions:

 ```
func fileExists(t *testing.T, path string, dir bool) bool {
  return exists(t, path, false)
}
func directoryExists(t *testing.T, path string, dir bool) bool {
  return exists(t, path, true)
}
```

Or you could just define some consts to `true`/`false`.